### PR TITLE
Doc + cleanup Pack class

### PR DESCRIPTION
**Pull request by firstred**
| Questions | Answers |
| --- | --- |
| Branch? | `develop` |
| Description? | Upgrading the PHPDoc comments of this class, using the latest PS standards and also applying the PS standards to some of the code. Also getting the Franglais out of the comments, getting rid of things the Symfony code sniffer nags about, etc. It might have some additional `todo` tags, depending on whether the code was comprehensible or not. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | N/A |
| How to test? | N/A |

Warning: this PR also has a code cleanup. Please check this first before merging!

### CODE REVIEW

File: classes/Pack.php [hunk 1]
```diff
@@ -353,16 +418,17 @@ public static function isCurrentlyUsed($table = null, $has_active_column = false
     /**
      * For a given pack, tells if it has at least one product using the advanced stock management
      *
-     * @param int $id_product id_pack
-     * @return bool
+     * @param int $idProduct id_pack
+     *
+*@return bool
```
Comment (mickaelandrieu) [inline classes/Pack.php]: you may have forgot some extra space, isn't it ?

File: classes/Pack.php [hunk 2]
```diff
@@ -376,16 +442,17 @@ public static function usesAdvancedStockManagement($id_product)
     /**
      * For a given pack, tells if all products using the advanced stock management
      *
-     * @param int $id_product id_pack
-     * @return bool
+     * @param int $idProduct id_pack
+     *
+*@return bool
```
Comment (mickaelandrieu) [inline classes/Pack.php]: same here

File: classes/Pack.php [hunk 3]
```diff
@@ -399,30 +466,32 @@ public static function allUsesAdvancedStockManagement($id_product)
     /**
      * Returns Packs that conatins the given product in the right declinaison.
      *
-     * @param integer $id_item Product item id that could be contained in a|many pack(s)
-     * @param integer $id_attribute_item The declinaison of the product
-     * @param integer $id_lang
+     * @param integer $idItem          Product item id that could be contained in a|many pack(s)
```
Comment (mickaelandrieu) [inline classes/Pack.php]: I think it's "int", not "integer"

File: classes/Pack.php [hunk 4]
```diff
@@ -33,94 +36,108 @@ class PackCore extends Product
     /**
      * Is product a pack?
      *
-     * @param $id_product
-     * @return bool
+     * @param $idProduct
```
Comment (mickaelandrieu) [inline classes/Pack.php]:
```diff
- * @param $idProduct
+ * @param int $idProduct
``

File: classes/Pack.php [hunk 5]
```diff
@@ -33,94 +36,108 @@ class PackCore extends Product
...
     }
 
     /**
      * Is product in a pack?
      * If $id_product_attribute specified, then will restrict search on the given combination,
      * else this method will match a product if at least one of all its combination is in a pack.
      *
-     * @param $id_product
-     * @param $id_product_attribute Optional combination of the product
+     * @param $idProduct
```
Comment (mickaelandrieu) [inline classes/Pack.php]: same here, would you mind to add some type information ?

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]
Good job! Some comments need to be adressed before :ship:

File: classes/Pack.php (hunk 1 above)
Comment (firstred) [inline classes/Pack.php]: Hey, that's a PhpStorm bug haha

**Comment by firstred** [comment]
Fixed

**Pull request closed without merging**